src/modules/numbers/numbers.service.ts

```diff
@@ -64,4 +66,41 @@ export class NumbersService {
       requestingSystem: dbNumber.REQUESTING_SYSTEM,
     };
   }
+
+  /**
+   * Helper to sort new ids
+   */
+  sortIds(newUkefIds: UkefId[]): UkefId[] {
+    const sortedIds = [];
+    const newIdsByType = this.groupIdsByType(newUkefIds);
+    const sortednewIdsByType = this.sortGroupedByTypeIds(newIdsByType);
+
+    // Keep result objects order, but ensure field maskedId is sorted.
+    newUkefIds.forEach((newId) => {
+      newId['maskedId'] = sortednewIdsByType[newId['type']].shift();
+      sortedIds.push(newId);
+    });
+    return sortedIds;
+  }
+
+  /**
+   * Helper to sort new Ids inside type array.
+   */
+  sortGroupedByTypeIds(newIdsByType: unknown): unknown {
```

Comment:
Should this be moved to a helper function? we will also need a unit test for it to make sure it works as expected

Reply:
Never mind, just seen the tests now - all good on that front

Reply:
I am not expecting to reuse this function and service class looks like a good place for it. Do we still want to move it out?

Reply:
Ok. Let's leave it as it is for now